apps/api/src/core/services/hono-error-handler/hono-error-handler.service.ts

```diff
@@ -25,6 +32,36 @@ export class HonoErrorHandlerService {
       return c.json({ error: "BadRequestError", data: error.errors }, { status: 400 });
     }
 
+    await this.reportError(error, c);
+
     return c.json({ error: "InternalServerError" }, { status: 500 });
   }
+
+  private async reportError<E extends Env = any>(error: Error, c: Context<E>): Promise<void> {
+    const id = this.sentry.captureEvent(await this.getSentryEvent(error, c));
+    this.logger.info({ event: "SENTRY_EVENT_REPORTED", id });
+  }
+
+  private async getSentryEvent<E extends Env = any>(error: Error, c: Context<E>): Promise<Event> {
+    const event = this.sentry.addRequestDataToEvent(this.sentryEventService.toEvent(error), {
+      method: c.req.method,
+      url: c.req.url,
+      headers: Object.fromEntries(c.req.raw.headers),
```

Comment:
Should we strip out sensitive headers like the `x-anonymous-user-id`?

Reply:
I'd leave it there on purpose actually. should help with debugging. do u see any issues with this?

Reply:
Well it's kind of a username/password so exposing it in sentry increases the risk. Do we have any other ID that we could use to identify the anonymous user without being able to login with it?

Reply:
imho we shouldn't go crazy trying to protect anonymous user accounts. I don't see any potential risks there. however, I can filter that out just in case. tracing info should be sufficient.

Reply:
discussed that it still makes sense to omit. fixed